<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>InviteDialog</class>
 <widget class="QDialog" name="InviteDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>450</width>
    <height>450</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Invite</string>
  </property>
  <property name="modal">
   <bool>true</bool>
  </property>
  <layout class="QVBoxLayout" name="inviteDialogLayout">
   <item>
    <widget class="QStackedWidget" name="pages">
     <widget class="QTabWidget" name="tabs">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="urlPage">
       <property name="currentIndex" stdset="0">
        <number>0</number>
       </property>
       <attribute name="title">
        <string>Session</string>
       </attribute>
       <layout class="QVBoxLayout" name="urlPageLayout">
        <item>
         <widget class="QPushButton" name="copyButton">
          <property name="text">
           <string>Copy Invite Link</string>
          </property>
          <property name="icon">
           <iconset theme="edit-copy">
            <normaloff>theme:edit-copy.svg</normaloff>theme:edit-copy.svg</iconset>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPlainTextEdit" name="urlEdit">
          <property name="maximumSize">
           <size>
            <width>16777215</width>
            <height>48</height>
           </size>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="includePasswordBox">
          <property name="text">
           <string>Include password</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="urlTopSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Expanding</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>0</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QLabel" name="localExplanationLabel">
          <property name="text">
           <string>You're hosting a session locally. If others can't join your session, take a look at &lt;a href=&quot;https://drawpile.net/localhosthelp&quot;&gt;the help page on hosting&lt;/a&gt;.</string>
          </property>
          <property name="textFormat">
           <enum>Qt::RichText</enum>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
          <property name="openExternalLinks">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="urlBottomSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>0</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QStackedWidget" name="codes">
       <attribute name="title">
        <string>Codes</string>
       </attribute>
       <widget class="QWidget" name="codesPage">
        <layout class="QVBoxLayout" name="codesPageLayout">
         <item>
          <widget class="widgets::SpanAwareTreeView" name="codesView">
           <property name="contextMenuPolicy">
            <enum>Qt::CustomContextMenu</enum>
           </property>
           <property name="selectionMode">
            <enum>QAbstractItemView::ContiguousSelection</enum>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="codeButtonLayout">
           <item>
            <widget class="QLabel" name="codeExplanationLabel">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string notr="true">What are invite codes?</string>
             </property>
             <property name="textFormat">
              <enum>Qt::RichText</enum>
             </property>
             <property name="wordWrap">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="createCodeButton">
             <property name="text">
              <string>Create</string>
             </property>
             <property name="icon">
              <iconset theme="list-add">
               <normaloff>theme:list-add.svg</normaloff>theme:list-add.svg</iconset>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="removeCodeButton">
             <property name="text">
              <string>Revoke</string>
             </property>
             <property name="icon">
              <iconset theme="list-remove">
               <normaloff>theme:list-remove.svg</normaloff>theme:list-remove.svg</iconset>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="codeLinkButton">
             <property name="text">
              <string>Copy</string>
             </property>
             <property name="icon">
              <iconset theme="edit-copy">
               <normaloff>theme:edit-copy.svg</normaloff>theme:edit-copy.svg</iconset>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
       <widget class="QWidget" name="noCodesPage">
        <layout class="QVBoxLayout" name="noCodesPageLayout">
         <item>
          <widget class="QLabel" name="noCodesExplanation">
           <property name="text">
            <string notr="true">No codes available explanation goes here.</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </widget>
     <widget class="QWidget" name="ipPage">
      <layout class="QVBoxLayout" name="ipPageLayout">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QLabel" name="ipExplanationLabel">
         <property name="text">
          <string>You're hosting a session locally, so Drawpile has to find your externally visible IP address to give you an invite link. For more information, check out &lt;a href=&quot;https://drawpile.net/localhosthelp&quot;&gt;the help page on hosting&lt;/a&gt;.</string>
         </property>
         <property name="textFormat">
          <enum>Qt::RichText</enum>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
         <property name="openExternalLinks">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="ipTopSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Preferred</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>24</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="ipButton">
         <property name="text">
          <string>Get externally visible IP address</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QProgressBar" name="ipProgressBar">
         <property name="maximum">
          <number>0</number>
         </property>
         <property name="value">
          <number>-1</number>
         </property>
         <property name="textVisible">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="ipBottomSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="enableCodesBox">
     <property name="text">
      <string>Allow session operators to manage invite codes</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QCheckBox" name="showOnHostBox">
     <property name="text">
      <string>Show this dialog when hosting a session</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttons">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Close</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>widgets::SpanAwareTreeView</class>
   <extends>QTreeView</extends>
   <header>desktop/widgets/spanawaretreeview.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>buttons</sender>
   <signal>accepted()</signal>
   <receiver>InviteDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttons</sender>
   <signal>rejected()</signal>
   <receiver>InviteDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
